src/array.jl

```diff
@@ -343,8 +343,15 @@ function _convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N};
     else
         # if order is defined for level type, automatically apply it
         L = leveltype(res)
-        if hasmethod(isless, Tuple{L, L})
+        if Base.OrderStyle(L) isa Base.Ordered
             levels!(res, sort(CategoricalArrays.levels(res)))
+        elseif hasmethod(isless, (L, L))
```

Comment:
I’d be tempted to drop this part on 1.6 and later, and rely on `OrderStyle` being `Ordered` for strings thanks to an upcoming PR.

Reply:
what about `elseif (VERSION < v"1.6.0") && (hasmethod(isles, (L,L))`?

Reply:
Yes, something like that.

Reply:
done